### ISSUE TITLE
[FR] Let the CLI report TX details (gas price, total cost, TX ID) when submitting a TX

### ISSUE THREAD

**Issue by jclapis** [rocket-pool/smartnode#129]
This is requst for an improvement to the CLI UX when the user invokes a command that submits a transaction to the eth network. The current UX has two problems:

1. The user doesn't get to see how much the TX costs based on their current gas price setting, so it can come as somewhat of a shock to run a transaction (e.g. creating a minipool) and learn that it just cost you 0.2 ETH on mainnet.
2. The user doesn't have any feedback after confirming the transaction; the CLI just hangs and waits for it to be mined. On mainnet with the wrong gas settings this can take *hours*.

The feature request here is to submit gas and transaction details, then either wait for the transaction to be mined or let the user exit after the transaction has been posted so they can do other things.

We envision that the UX could take two forms:

A
```
...
Are you sure you want to stake 160.000000 RPL? Staked RPL can only be withdrawn after a delay.
y

With your current gas price setting of 2 Gwei, this will cost up to 0.000513844 ETH in gas fees.
Do you want to continue? [y/n]
y

Transaction has been submitted with hash 0xb3680215cd509df88baf4fd89260e4c0b6bc676669325853dbd6413ec34e252c.
You may follow its progress by visiting:
https://goerli.etherscan.io/tx/0xb3680215cd509df88baf4fd89260e4c0b6bc676669325853dbd6413ec34e252c

Waiting for the transaction to be mined... you may wait here for it, or press q to exit and return to the terminal.

Successfully staked 160.000000 RPL.
```

B
```
Are you sure you want to stake 160.000000 RPL? Please note that:

  * Staked RPL can only be withdrawn after a delay.
  * With your current gas price setting of 2 Gwei, this will cost up to 0.000513844 ETH in gas fees.

Do you want to continue? [y/n]
y

Transaction has been submitted with hash 0xb3680215cd509df88baf4fd89260e4c0b6bc676669325853dbd6413ec34e252c.
You may follow its progress by visiting:
https://goerli.etherscan.io/tx/0xb3680215cd509df88baf4fd89260e4c0b6bc676669325853dbd6413ec34e252c

Waiting for the transaction to be mined... you may wait here for it, or press q to exit and return to the terminal.

Successfully staked 160.000000 RPL.
```

*([See here](https://discord.com/channels/405159462932971535/704214707904446535/830544837500469309) for the discussion.)*

Though opinions on which one is best differ, either case seems to be preferable to the current UX according to Discord.

**Comment by LookingForOwls** [comment]
In the original discussion I was leaning towards two separate prompts but I now think the second option is better. 

The two key points are presented well and most users should have no issue reviewing them before agreeing or not.

**Comment by kidkal** [comment]
I've done some work towards estimating gas before proceeding https://github.com/rocket-pool/smartnode/pull/98  was done for deposit step since that took the most gas but can be applicable to all transactions.

**Comment by jclapis** [comment]
Implemented in #133 and #146.
